### PR TITLE
[mlir][openacc] Add legalize data pass for compute operation

### PR DESCRIPTION
**Pull request by clementval**
This patch adds a simple pass to replace the uses inside compute operation. It replaces the `varPtr` values with their corresponding `accPtr` values gathered through the dataClauseOperands.

private and reductions variables are not included in this pass since they will normally be replace when they are materialized.

### CODE REVIEW

**Review by razvanlupusoru — APPROVED** [review]
Looks great! Implementation looks much cleaner than I imagined.

I do think that the ssa uses for private/reduction/firstprivate operations should also be replaced. This means that the logic would also need to apply to acc.loop in addition to the compute constructs. This should be added also under an option just like you did for hostToDevice.

I would say this is OK to submit as-is - and we can make the improvement when it is needed. Nice work! Thank you for getting this done :)

File: mlir/include/mlir/Dialect/OpenACC/Transforms/Passes.td [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+#ifndef MLIR_DIALECT_OPENACC_TRANSFORMS_PASSES
+#define MLIR_DIALECT_OPENACC_TRANSFORMS_PASSES
+
+include "mlir/Pass/PassBase.td"
+
+def LegalizeDataInRegion : Pass<"openacc-legalize-data", "mlir::func::FuncOp"> {
+  let summary = "Legalize the data in the compute region";
+  let description = [{
+    This pass replace uses of varPtr in the compute region with their accPtr
```
Comment (vzakhari) [inline mlir/include/mlir/Dialect/OpenACC/Transforms/Passes.td:17]: nit
```suggestion
    This pass replaces uses of varPtr in the compute region with their accPtr
```

**Review by vzakhari — APPROVED** [review]
Looks great!

**Comment by clementval** [comment]
> Looks great! Implementation looks much cleaner than I imagined.
> 
> I do think that the ssa uses for private/reduction/firstprivate operations should also be replaced. This means that the logic would also need to apply to acc.loop in addition to the compute constructs. This should be added also under an option just like you did for hostToDevice.
> 
> I would say this is OK to submit as-is - and we can make the improvement when it is needed. Nice work! Thank you for getting this done :)

I'm gonna make a follow up PR for private/reduction

**Pull request merged**